dataactvalidator/sqs_work_dispatcher.py

```diff
@@ -0,0 +1,706 @@
+            # Then execute the given function with the given args
+            func(*args)
+
+        self._worker_process = ctx.Process(
+            name=self.worker_process_name,
+            target=signal_reset_wrapper,
+            args=(job, job_args),
+            daemon=True  # daemon=True ensures that if the parent dispatcher dies, it will kill this child worker
+        )
+        self._worker_process.start()
+        log_job_message(
+            logger=self._logger,
+            message="Worker process named [{}] started with process ID [{}]".format(
+                self.worker_process_name,
+                self._worker_process.pid
+            ),
+            is_debug=True
+        )
+        self._monitor_work_progress()
+        return True
+
+    def dispatch(self, job, message_transformer=lambda x: x.body, additional_job_args=(),
+                 worker_process_name=None, exit_handler=None):
+        """
+        Get work from the queue and dispatch it in a newly starter worker process
+
+        Poll the queue for a message to work on. Combine that message with any
+        _additional_job_args passed into the constructor, and feed this complete set of arguments to the provided
+        callable job that will be executed in a separate worker process. The worker process will be monitored so that
+        heartbeats can be sent to the SQS to allow it to keep going.
+
+        :param job: the callable to execute as work for the job
+        :param message_transformer: A lambda to extract arguments to be passed to the callable from the queue message.
+               By default, a function that just returns the message body is used.
+        :param additional_job_args: Additional arguments to provide to the callable, along with those from the message
+        :param worker_process_name: Name given to the newly created child process. If not already set, defaults to
+               the name of the provided job callable
+        :param exit_handler: a callable to be called when handling an `EXIT_SIGNAL` signal, giving the
+               opportunity to perform cleanup before the process exits. Gets the job_args passed to it when run
+        :raises SystemExit(1): If it can't connect to the queue or receive messages
+        :raises QueueWorkerProcessError: Under various conditions where the child worker process fails to run the job
+        :raises QueueWorkDispatcherError: Under various conditions where the parent dispatcher process can't
+                orchestrate work execution, monitoring, or exit-signal-handling
+        :return: True if a message was found on the queue and dispatched, otherwise False if nothing on the queue
+        """
+        self._dequeue_message(self._long_poll_seconds)
+        if self._current_sqs_message is None:
+            return False
+        msg_args = message_transformer(self._current_sqs_message)
+        if isinstance(msg_args, list) or isinstance(msg_args, tuple):
```

Comment:
I think it may be doable, and I kindof like the idea ... but it has a pretty big ripple-effect throughout the code, which increases the risk of a change like this:
- need to change the call of the [`multiprocessing.Process()` `__init__`](https://docs.python.org/3.5/library/multiprocessing.html#multiprocessing.Process) to use the `kwargs` argument instead of `args` at the central call to it
- Would also need to change the wrapper function that does the resets of the signal to pass in a dict/kwargs instead of tuple
- Would need to change the private `_dispatch` and public `_dispatch*` methods to accept args as a dict
- Would need to refactor all unit tests to send in the args as a dict

Then it'd be best to go through the integration testing again in sandbox to make sure there's no regression since these changes are subtle. 

What do you think?

Reply:
I personally think it's worth it preventing the potential confusion. I could help out if need be.

Reply:
Done. 
See changes: https://github.com/fedspendingtransparency/data-act-broker-backend/pull/1568/files#diff-c8d788bd3085ca32d20adb7af4bab886R78

And examples:
- https://github.com/fedspendingtransparency/data-act-broker-backend/pull/1568/files#diff-39b84be5ae703d5f2a2add3a76ebb0f7R234
- https://github.com/fedspendingtransparency/data-act-broker-backend/pull/1568/files#diff-39b84be5ae703d5f2a2add3a76ebb0f7R309